great_expectations/cli/toolkit.py

```diff
@@ -699,8 +706,9 @@ def upgrade_project_zero_versions_increment(
         update_version=False,
         from_cli_upgrade_command=from_cli_upgrade_command,
     )
+    context: Optional[DataContext]  # type: ignore[no-redef] # allow re-def here
     if exception_occurred or increment_version:
-        context = None
+        context = None  # type: ignore[assignment] # see redef above
```

Comment:
Why `return None` above and then assign `None` here?

Reply:
In both cases the early `return` means I wouldn't have to do a `type: ignore`, but here the return statements would be far apart which I though hurt readability.

But I could invert the conditions here and put them next to each other, and then I wouldn't have to ignore errors.

Reply:
I ended up just returning early. Inverting the conditions made for some kinda hard to read `if not exception_occurred and not icrement_version:`  conditional.
But if folks prefer this I can do that instead.